src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,12 +8,16 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_ERROR = "Missing arguments for address";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
     private boolean isPrivate;
+    public final Block block;
+    public final Street street;
+    public final Unit unit;
+    public final PostalCode postalCode;
 
```

Comment:
Shouldn't they be private and be accessed only by accessor methods? This current  design violates encapsulation.